integration/testdata/snyk/snyk-pipeline.bubbly

```diff
@@ -0,0 +1,166 @@
+resource "extract" "snyk" {
+    api_version = "v1"
+    spec {
+        input "file" { }
+        type = "json"
+        source {
+            file = self.input.file
+            format = object({
+                vulnerabilities: list(object({
+                    CVSSv3: string,
+                    alternativeIds: list(string),
+                    creationTime: string,
+                    credit: list(string),
+                    cvssScore: number,
+                    description: string,
+                    exploit: string,
+                    fixedIn: list(string),
+                    functions: list(string),
+                    functions_new: list(string),
+                    id: string,
+                    identifiers: object({
+                        CVE: list(string),
+                        CWE: list(string),
+                    }),
+                    language: string,
+                    modificationTime: string,
+                    moduleName: string,
+                    packageManager: string,
+                    packageName: string,
+                    patches: list(string),
+                    proprietary: bool,
+                    publicationTime: string,
+                    references: list(object({
+                        title: string,
+                        url: string,
+                    })),
+                    semver: object({
+                        hashesRange: list(string),
+                        vulnerable: list(string),
+                    }),
+                    severity: string,
+                    severityWithCritical: string,
+                    title: string,
+                    from: list(string),
+                    upgradePath: list(string),
+                    isUpgradable: bool,
+                    isPatchable: bool,
+                    name: string,
+                    version: string,
+                }))
+            })
```

Comment:
Imagine a world in which you didn't have to write the `format`? 😏

I think it might be possible, if we leverage reflection.